py/private/pytest_shard/pytest_shard.py

```diff
@@ -0,0 +1,73 @@
+"""
+This has been adapted from the original code at
+https://github.com/AdamGleave/pytest-shard
+The main difference is that the plugin has been converted to a class
+so that it can be passed to pytest.main's plugins attribute, and the
+shard selection was changed from a hash-based approach to a round-robin
+approach for a more balanced distribution with smaller numbers of tests.
+"""
+
+from typing import Iterable, List, Sequence
+
+from _pytest import nodes  # for type checking only
+
+
+def positive_int(x) -> int:
+    x = int(x)
+    if x < 0:
+        raise ValueError(f"Argument {x} must be positive")
+    return x
+
+
+def filter_items_by_shard(
+    items: Iterable[nodes.Node], shard_id: int, num_shards: int
+) -> Sequence[nodes.Node]:
+    """Computes `items` that should be tested in `shard_id` out of `num_shards` total shards."""
+    shards = [i % num_shards for i in range(len(items))]
+
+    new_items = []
+    for shard, item in zip(shards, items):
+        if shard == shard_id:
+            new_items.append(item)
+    return new_items
+
+
+class ShardPlugin:
+    @staticmethod
+    def pytest_addoption(parser):
+        """Add pytest-shard specific configuration parameters."""
+        group = parser.getgroup("shard")
+        group.addoption(
+            "--shard-id",
+            dest="shard_id",
+            type=positive_int,
+            default=0,
+            help="Number of this shard.",
+        )
+        group.addoption(
+            "--num-shards",
+            dest="num_shards",
+            type=positive_int,
+            default=1,
+            help="Total number of shards.",
+        )
+
+    @staticmethod
+    def pytest_report_collectionfinish(config, items: Sequence[nodes.Node]) -> str:
+        """Log how many and, if verbose, which items are tested in this shard."""
+        msg = f"Running {len(items)} items in this shard"
+        if config.option.verbose > 0 and config.getoption("num_shards") > 1:
```

Comment:
how does one set this verbose option?

Reply:
It looks like we already do https://github.com/aspect-build/rules_py/pull/493/files#diff-eab700c56364dc4619c882e7e89bf26cd4cd1cd9efe61284e3bd0607bfe152bcR31